src/plot_api/subroutines.js

```diff
@@ -397,24 +399,98 @@ function findCounterAxisLineWidth(ax, side, counterAx, axList) {
 }
 
 exports.drawMainTitle = function(gd) {
+    var title = gd._fullLayout.title;
+    setDflts(title);
     var fullLayout = gd._fullLayout;
-
     var textAnchor = getMainTitleTextAnchor(fullLayout);
     var dy = getMainTitleDy(fullLayout);
+    var y = getMainTitleY(fullLayout, dy);
+    var position = fullLayout.title.y > 0.5 ? 't' : 'b';
+
+    if(gd._fullLayout.title.automargin) {
```

Comment:
`title` is declared here. So we can rewrite:
```suggestion
    if(title.automargin) {
```